src/openforms/submissions/api/permissions.py

```diff
@@ -67,7 +68,14 @@ def has_object_permission(self, request: Request, view: APIView, obj) -> bool:
         submission_url_kwarg = getattr(view, "submission_url_kwarg", default_url_kwarg)
 
         submission_uuid = view.kwargs[submission_url_kwarg]
-        return owns_submission(request, submission_uuid)
+        submission = (
+            obj
+            if isinstance(obj, Submission)
+            else Submission.objects.get(uuid=submission_uuid)
+        )
+        return owns_submission(request, submission_uuid) and meets_plugin_requirements(
+            request, submission.form.authentication_backend_options
```

Comment:
I'd also `select_related` the `form` in the query or use a `.values('form__authentication_backend_options')` query for a more performant query. Considering this is a permission class that's used in most endpoints, :zap:  is nice.